presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -315,34 +361,44 @@ public FragmentProperties setSingleNodeDistribution()
             return this;
         }
 
-        public FragmentProperties setDistribution(PartitioningHandle distribution)
+        public FragmentProperties setDistribution(PartitioningHandle distribution, Metadata metadata, Session session)
```

Comment:
What about call it `coalesceDistribution`?  Or in the comment mention it now will coalesce fragment distribution to a common one